File: LeetCode/00206. Reverse Linked List/00206.c
// https://leetcode.com/problems/reverse-linked-list

struct ListNode* reverseList(struct ListNode *head) {
    struct ListNode *prev = NULL, *next = NULL;

    while (head != NULL) {
        next = head->next;
        head->next = prev;
        prev = head;
        head = next;
    }

    return prev;
}

File: LeetCode/00206. Reverse Linked List/00206.go
// https://leetcode.com/problems/reverse-linked-list

func reverseList(head *ListNode) *ListNode {
    dummy := ListNode{}

    for head != nil {
        curr := head
        head = head.Next
        curr.Next = dummy.Next
        dummy.Next = curr
    }

    return dummy.Next
}
